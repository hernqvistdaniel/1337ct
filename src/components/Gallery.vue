<script lang="ts">
import GalleryCard from './GalleryCard.vue';

export default {
  components: { GalleryCard },
  props: ['filter', 'presentations', 'sort'],
  data() {
    return {
      filteredPres: null,
      sortedPres: null
    }
  },
  methods: {
    changeFilter() {
      console.log(this.filteredPres)
      this.filteredPres = this.presentations.filter(pres => {
        if (pres.office) {
          return pres.office.toLowerCase() == this.filter.toLowerCase()
        }
      })
      console.log(this.filteredPres)
    },
    changeSort() {
      console.log(this.sortedPres)
      this.sortedPres = this.presentations.sort((a, b) => {
        if (a.phoneNumber && b.phoneNumber) {
          return a.phoneNumber.localeCompare(b.phoneNumber)
        }
      })

    }
  },
  watch: {
    filter() {
      this.changeFilter()
    },
    sort() {
      this.changeSort()
    }
  },
  mounted() {
    console.log(this.presentations)
    this.filteredPres = this.presentations
    this.sortedPres = this.presentations
  }
}
</script>

<template>
  <div class="cardContainer">
    <GalleryCard v-for="(presentation, index) in filteredPres" :presentation="presentation" :id="index" />
  </div>
</template>

<style scoped>
.cardContainer {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
</style>
