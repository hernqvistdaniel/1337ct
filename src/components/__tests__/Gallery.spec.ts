import { describe, it, expect, vi } from 'vitest'
import { createStore, Store } from 'vuex'
import { shallowMount } from '@vue/test-utils'
import Gallery from '../Gallery.vue'

describe('Gallery.vue', () => {
  const store = createStore({
    state() {
      return {
        presentations: [{
          "name": "Akan Murat Cimen",
          "email": "akan.murat.cimen@1337.tech",
          "phoneNumber": "076-1337198",
          "office": "Lund",
          "manager": "jim@1337.tech",
          "orgUnit": "/Employees",
          "mainText": "<p>It has always been appealing to me to try to understand everything from the big bang to AI-suggested cat videos by asking the what, why, and how questions. If I ask the right questions at the right time, in the right order, I believe everything has an answer. The knowledge we have learned as human beings so far by asking these questions is an invaluable treasure.</p><p>I have been demanding some operations from CPUs since I was a kid. I worked on various projects like e-commerce, content management systems, car sharing, live streaming etc. with various programming languages. I think the true \"legacy code\" is not about how old it is, it's about its readability, maintainability etc. I'm trying not to be that person who writes code like that.</p><p>In my spare time I'm demanding more operations from CPUs by playing games, watching movies, tv-series or traveling with the help of a navigation application.</p> ",
          "gitHub": "akanmuratcimen",
          "twitter": null,
          "stackOverflow": "513234",
          "linkedIn": "/in/akanmuratcimen/",
          "imagePortraitUrl": "https://i.1337co.de/profile/akan-murat-cimen",
          "imageWallOfLeetUrl": "https://i.1337co.de/wallofleet/akan-murat-cimen",
          "highlighted": false,
          "published": true
      },]
      }
    },
  })
  const wrapper = shallowMount(Gallery, {store})

  it('renders properly', () => {
    expect(wrapper.html()).toBeTruthy()
  })

  it('displays the card container', () => {
    const cardcontainer = wrapper.find('.cardContainer')
    expect(cardcontainer.html()).toBeTruthy()
  })

  it('matches a snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
