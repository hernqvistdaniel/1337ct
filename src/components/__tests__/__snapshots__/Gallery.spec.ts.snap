// Vitest Snapshot v1

exports[`Gallery.vue > matches a snapshot 1`] = `"<div class=\\"cardContainer\\" data-v-15c1b052=\\"\\"></div>"`;
